### PR TITLE
feat: support setting up existing container as devcontainer

### LINKED ISSUE #782: Setup dev container in an existing running container
**Is your feature request related to a problem?**
I sometimes have a running container with lots of resources and I want to use that as up my devcontainer in that.

**Which solution do you suggest?**
Add a command `set-up` or a sub-command to `up` to start a workspace in an existing container. My guess is that this just needs to be exposed and most, if not all features are already present in the code base.

**Which alternative solutions exist?**
`devcontainer` CLI provides `devcontainer set-up` where you can pass a container ID to set it up as a devcontainer.

Setup existing container as devcontainer using devcontainer CLI

**Additional context**
I looked at `devpod agent container setup` and it's description kind of says the same thing, but I'm not sure if it is supposed to be used directly or not. Also, I am unsure about what to pass in the `--setup-info` and `--workspace-info` options to make it work.

devcontainer agent container setup command help output

### PR DESCRIPTION
**Pull request by amitds1997**
Closes #782.

### CODE REVIEW

**Comment by amitds1997** [comment]
Let me know if I should add to any existing test suite or any other tests I should update. I could not find one.

To run `DevPod` in a container, use the following command syntax: `devpod up <workspace-name> --source container:<container-id> <other-devpod-options>`

`<container-id>` can be the container ID or the container name. Container ID is preferred since it is unique.
`<other-devpod-options>` can be any up options except `--recreate`.

### `devpod stop` and `devpod delete` behavior

1. For `devpod delete`, we **do not delete** the container, since it was not created and launched by DevPod.
2. For `devpod stop`, we **do stop** the container since it is now set up as a devcontainer, so it should be okay to stop the container since it can easily be restarted too.

File: pkg/docker/helper.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 
 type DockerHelper struct {
 	DockerCommand string
+	ContainerID   string
```
Comment (pascalbreuninger) [inline pkg/docker/helper.go:22]: could you add a comment explaining the purpose of this field please? Might not be clear in a couple of months

Comment (amitds1997) [reply pkg/docker/helper.go]: Done in [499813a](https://github.com/loft-sh/devpod/pull/839/commits/499813a6d5846706879ea9f7d51b5762170a6e5f). Let me know if we want to add more details there

Comment (pascalbreuninger) [reply pkg/docker/helper.go]: great, thanks

**Comment by pascalbreuninger** [comment]
@amitds1997 we have a failing test, could you have a look at that please?

**Comment by pascalbreuninger** [comment]
> > @amitds1997 we have a failing test, could you have a look at that please?
> 
> Seems like an issue with pulling docker image ([log link](https://github.com/loft-sh/devpod/actions/runs/7389348372/job/20128487008?pr=839#step:7:45)). Might this be an intermittent thing? The workflows on the previous commit all passed (including the [failing one](https://github.com/loft-sh/devpod/actions/runs/7302408927/job/20084623169)). Since then, I have only added a comment. So, nothing should break.

yeah I thought the same earlier, that's why I re-ran the tests. Let's try again tomorrow, maybe it'll sort itself out 🤷‍♂️ 
 
> Nonetheless, I am trying to reproduce this locally, in case something is amiss. I would go ahead and add an e2e test for launching devpod with a running container.

that'd be great!

**Review by pascalbreuninger — APPROVED** [review]
Let's get this merged, thanks a lot @amitds1997!

**Comment by amitds1997** [comment]
Hey @pascalbreuninger, when are we planning to send out a release with these changes? Are there going to be more alpha releases? Thanks!

**Comment by pascalbreuninger** [comment]
we can cut an alpha now, was planning to do a stable one towards the end of the month

**Comment by hrittikhere** [comment]
BTW Pascal, do we have any documentation for this feature?

**Pull request merged**